### ISSUE TITLE
xpath.clj arguments contains the script name

### ISSUE THREAD

**Issue by lkuty** [pjt/saxon#1]
Hi,

The variable _command-line-args_ contains the name of the program at first position. So we have to take the rest of the sequence to get the right argument list.

(let [args _command-line-args_

->

(let [args (rest _command-line-args_)

Ludovic Kuty

**Comment by lkuty** [comment]
The asterisks surrounding command-line-args have disappeared and the text appears in italic. First time I use github, so I didn't know it would do that.

**Comment by lkuty** [comment]
OK this my own fault. My clj bash script used to launch xpath.clj isn't properly written.

**Comment by pjt** [comment]
Hi Ludovic,

Thanks for the message — the value of *command-line-args* does depend on your `clj` script, as you've found out. I believe most use (something like)

  `exec java $OPTS -cp $CLASSPATH clojure.main "$@"`

but there's no standard script in the distro, so it's easy to have inconsistency in the community. Clojure-contrib does have an example Bash launcher script, which runs off of enviro variables: it's [here](http://github.com/richhickey/clojure-contrib/blob/master/launchers/bash/clj-env-dir).

Let me know how the saxon lib works out for you. In case you've started to use leiningen, check out the `lein` branch, which keeps the Saxon Java depencies in the clojars.org repository. 

Best,
Perry

**Comment by lkuty** [comment]
Hi Perry,

Thanks for your reply. I took the clj script from http://en.wikibooks.org/wiki/Clojure_Programming/Getting_Started#Enhancing_Clojure_REPL_with_rlwrap but their script is not correct.
I ended up with this:
    java -cp $CLOJURE_JAR:$CLOJURE_CONTRIB_JAR:$SAXON_JAR clojure.main "$@"
As you wrote, the trick was to add $@ between quotes and to avoid adding an extra $1 as they've done in their script.
I will look at the Bash launcher script. I just tried to run xpath.clj with some XPath 2.0 expressions and it ran fine. My goal is to make all sort of XML processing in Clojure that were previously done in Java. 
I didn't know about leiningen and it looks interesting. Thanks again for it :)

Regards,

Ludovic
